tests/test_pattern.py

```diff
@@ -49,6 +50,10 @@ def choice(self, _outcomes: list[Outcome]) -> Outcome:
         return next(self.__it)
 
 
+def try_use_rustworkx() -> bool:
+    return util.find_spec("rustworkx") is not None
```

Comment:
@shinich1 In the first place, I'm not sure why `rustworkx` was made optional. Do you remember why?

Reply:
@EarlMilktea actually that's a good point! When we first added rustworkx did not support some combinations of OS/version, thus this impl. If rustworkx is much more mature now, we can just drop networkx version. What do you think?

Reply:
@shinich1 

It would greatly simplify the code. I'm for that.

Reply:
I also would like to do thorough refactor.

Reply:
On the other hand, `networkx` has the nice property to be pure Python, and it may worth keep it for portability reason. Moreover, being able to target multiple (two) graph back-ends help to have the right abstractions in `BaseGraphState`.

Reply:
@thierry-martinez @shinich1 
I'm not opposed to keep `networkx`, but at the same time I'm strongly tempted to drop `rxgraphviews`.
This module is forcibly making `rustworkx` behave like `networkx` by translating its internal representation.
This can never the right way.

Reply:
Additionally, let me point out that `networkx` has very poor type annotation support: we somehow need to isolate/wrap it to prevent `Any` or type warnings.

Reply:
@thierry-martinez 

> for portability

`fastflow` will anyway introduce similar portability constraints as `rustworkx`, but it would not be a problem because for such environments even `numpy` is not binary-distributed in the first place.
Also see https://github.com/TeamGraphix/fastflow/blob/2-implement-flow-algorithm/.github/workflows/wheel.yml .

Reply:
>  being able to target multiple (two) graph back-ends help to have the right abstractions in BaseGraphState.

@thierry-martinez do you have `stim` integration in mind regarding above (which you mentioned are ongoing on Paris side)?
I am tempted to drop `networkx`, as doing so should make it easier to maintain. If `stim` or other stabilizer simulator may come in to this, let us keep the abstract interface. what do you think? @EarlMilktea @thierry-martinez

Reply:
I don't think that we have to keep `networkx` for `stim`, but I believe that having the `networkx` back-end may have several advantages: beside keeping a pure Python implementation, it can be helpful for instance to catch bugs in other backends (see https://github.com/TeamGraphix/graphix/issues/206 ).